cordova-lib/src/util/promise-util.js

```diff
@@ -33,4 +33,21 @@ function Q_chainmap(args, func) {
     });
 }
 
+// Behaves similar to Q_chainmap but gracefully handles failures.
+// When a promise in the chain is rejected, it will call the failureCallback and then continue the processing, instead of stopping
```

Comment:
Nit: The line is twice as longer than previous one. It might look better if split into two lines.
Another nit: Since this method is 'public', the comment might be rewritten using JSDoc notation.